src/main/java/com/powsybl/openloadflow/OpenSecurityAnalysis.java

```diff
@@ -0,0 +1,474 @@
+        for (Contingency contingency : contingencies) {
+            ContingencyContext contingencyContext = new ContingencyContext(contingency);
+            contingencyContexts.add(contingencyContext);
+
+            Set<Switch> switchesToOpen = new HashSet<>();
+            for (ContingencyElement element : contingency.getElements()) {
+                switch (element.getType()) {
+                    case BRANCH:
+                        contingencyContext.branchIdsToOpen.add(element.getId());
+                        break;
+                    default:
+                        throw new UnsupportedOperationException("TODO");
+                }
+                AbstractTrippingTask task = element.toTask();
+                task.traverse(network, null, switchesToOpen, new HashSet<>());
+            }
+
+            for (Switch sw : switchesToOpen) {
+                contingencyContext.branchIdsToOpen.add(sw.getId());
+                allSwitchesToOpen.add(sw);
+            }
+        }
+        return contingencyContexts;
+    }
+
+    List<LfNetwork> createNetworks(Set<Switch> allSwitchesToOpen, AcLoadFlowParameters acParameters) {
+        List<LfNetwork> lfNetworks;
+        String tmpVariantId = "olf-tmp-" + UUID.randomUUID().toString();
+        network.getVariantManager().cloneVariant(network.getVariantManager().getWorkingVariantId(), tmpVariantId);
+        try {
+            for (Switch sw : network.getSwitches()) {
+                sw.setRetained(false);
+            }
+            for (Switch sw : allSwitchesToOpen) {
+                sw.setRetained(true);
+            }
+            lfNetworks = AcloadFlowEngine.createNetworks(network, acParameters);
+        } finally {
+            network.getVariantManager().removeVariant(tmpVariantId);
+        }
+        return lfNetworks;
+    }
+
+    /**
+     * Detect violations on branches and on buses
+     * @param branches branches on which the violation limits are checked
+     * @param buses buses on which the violation limits are checked
+     * @param violations list on which the violation limits encountered are added
+     */
+    private void detectViolations(Stream<LfBranch> branches, Stream<LfBus> buses, List<LimitViolation> violations) {
+        // Detect violation limits on branches
+        branches.forEach(b -> detectBranchViolations(b, violations));
+
+        // Detect violation limits on buses
+        // TODO
+    }
+
+    /**
+     * Detect violation limits on one branch and add them to the given list
+     * @param branch branch of interest
+     * @param violations list on which the violation limits encountered are added
+     */
+    private void detectBranchViolations(LfBranch branch, List<LimitViolation> violations) {
+        // Detect violation on the left
+        detectBranchViolationSide(branch.getId(), Branch.Side.ONE, branch.getBus1(), branch.getI1(), branch.getPermanentLimit1(), violations);
```

Comment:
I am not sure that we need this refactoring. In a first version, we just take into account permanent limit for current. We will then support temporary limits for current. After that, we will try to do the same for apparent power limit. I think that it is better to imagine the refactoring with the whole picture or do it at the end.

Reply:
````detectBranchViolationSide```` indeed I am talking about this function.

Reply:
We do not need this refactoring, sure. It was just meant to avoid code duplication and to improve readability, but as you disagree on that point I'll revert it.